src/libs/copy_history.c

```diff
@@ -311,15 +310,11 @@ static void paste_parts_button_clicked(GtkWidget *widget, gpointer user_data)
   // back dt_view_get_images_to_act_on() which free list and create a new one. So the
   // above imgs will be invalidated.
 
-  GList *l_copy = g_list_copy((GList *)imgs);
```

Comment:
I'm also really dubious about this one. See comment just above. Remember that dt_view_get_images_to_act_on() returns a list that is stored into the view. Maybe this is something to change, but see last line in this routine:

`  return darktable.view_manager->act_on.images;`

See also commit : 3afe650ce1ad768e805bf05db18c1e74967028a5.